cmake/sca/codechecker/sca.cmake

```
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Basalte bv
+
+find_program(CODECHECKER_EXE CodeChecker REQUIRED)
+message(STATUS "Found CodeChecker: ${CODECHECKER_EXE}")
+
+# CodeChecker uses the compile_commands.json as input
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# Create an output directory for our tool
+set(output_dir ${CMAKE_BINARY_DIR}/sca/codechecker)
+file(MAKE_DIRECTORY ${output_dir})
+
+# Use a dummy file to let CodeChecker know we can start analyzing
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands COMMAND
+  touch ${output_dir}/codechecker.ready)
+set_property(GLOBAL APPEND PROPERTY extra_post_build_byproducts
+  ${output_dir}/codechecker.ready)
+
+add_custom_target(codechecker ALL
+  COMMAND ${CODECHECKER_EXE} analyze
+    --keep-gcc-include-fixed
+    --keep-gcc-intrin
+    --output ${output_dir}/codechecker.plist
+    ${CODECHECKER_ANALYZE_OPTS}
+    ${CMAKE_BINARY_DIR}/compile_commands.json
+    || true
+  DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json ${output_dir}/codechecker.ready
+  BYPRODUCTS ${output_dir}/codechecker.plist
+  VERBATIM
+  USES_TERMINAL
+  COMMAND_EXPAND_LISTS
+)
+
+# Cleanup dummy file
+add_custom_command(
+  TARGET codechecker POST_BUILD
+  COMMAND rm ${output_dir}/codechecker.ready
```

Please use suggestion version to have an OS independent implementation
```suggestion
  COMMAND ${CMAKE_COMMAND} -E rm ${output_dir}/codechecker.ready
```